### PR TITLE
issue: 3790452 L2 cache miss optimization

### PR DESCRIPTION
**Pull request by galnoam**
## Description
L2 cache miss optimization

##### What
reorder struct members

##### Why ?
data locality

## Change type
What kind of change does this PR introduce?
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Tests
- [ ] Other

## Check list
- [ ] Code follows the style de facto guidelines of this project
- [ ] Comments have been inserted in hard to understand places
- [ ] Documentation has been updated (if necessary)
- [ ] Test has been added (if possible)

### CODE REVIEW

File: src/core/sock/socket_fd_api.h [hunk 1]
```diff
@@ -92,16 +92,17 @@ enum {
  */
 typedef struct xlio_tx_call_attr {
     tx_call_t opcode;
+    unsigned xlio_flags;
```
Comment (AlexanderGrissik) [inline src/core/sock/socket_fd_api.h:95]: Added as part of PR112

File: src/core/sock/socket_fd_api.h [hunk 2]
```diff
@@ -92,16 +92,17 @@ enum {
...
+
     struct _attr {
         struct iovec *iov;
         ssize_t sz_iov;
         int flags;
-        struct sockaddr *addr;
         socklen_t len;
+        struct sockaddr *addr;
```
Comment (AlexanderGrissik) [inline src/core/sock/socket_fd_api.h:102]: Added as part of PR112

File: src/core/sock/sockinfo.h [hunk 3]
```diff
@@ -560,6 +559,8 @@ class sockinfo : public socket_fd_api,
         std::deque<struct ring_ec> ec_cache;
         struct ring_ec *ec;
     } m_socketxtreme;
+    void *m_fd_context; // Context data stored with socket
```
Comment (AlexanderGrissik) [inline src/core/sock/sockinfo.h:562]: Added as part of PR112

File: src/core/sock/sockinfo.h [hunk 4]
```diff
@@ -595,11 +596,13 @@ class sockinfo : public socket_fd_api,
     rx_net_device_map_t m_rx_nd_map;
     rx_flow_map_t m_rx_flow_map;
     // we either listen on ALL system cqs or bound to the specific cq
-    ring *m_p_rx_ring; // used in TCP/UDP
-    buff_info_t m_rx_reuse_buff; // used in TCP instead of m_rx_ring_map
     bool m_rx_reuse_buf_pending; // used to periodically return buffers, even if threshold was not
                                  // reached
     bool m_rx_reuse_buf_postponed; // used to mark threshold was reached, but free was not done yet
+    ring *m_p_rx_ring; // used in TCP/UDP
```
Comment (AlexanderGrissik) [inline src/core/sock/sockinfo.h:602]: Added as part of PR112

File: src/core/sock/sockinfo.h [hunk 5]
```diff
@@ -595,11 +596,13 @@ class sockinfo : public socket_fd_api,
...
+    buff_info_t m_rx_reuse_buff; // used in TCP instead of m_rx_ring_map
```
Comment (AlexanderGrissik) [inline src/core/sock/sockinfo.h:603]: These should not be used by socketxtreme DP.

File: src/core/sock/sockinfo_tcp.h [hunk 6]
```diff
@@ -512,20 +512,25 @@ class sockinfo_tcp : public sockinfo, public timer_handler {
     list_node<sockinfo_tcp, sockinfo_tcp::accepted_conns_node_offset> accepted_conns_node;
 
 private:
+    lock_dummy m_tcp_con_lock;
```
Comment (AlexanderGrissik) [inline src/core/sock/sockinfo_tcp.h:515]: Currently we can't remove locks because some control path sockets should still be protected.

**Comment by galnoam** [comment]
.

**Pull request closed without merging**